packages/components/src/unit-control/utils.ts

```diff
@@ -318,36 +323,32 @@ export const useCustomUnits = ( {
  * accurately displayed in the UI, even if the intention is to hide
  * the availability of that unit.
  *
- * @param  currentValue Selected value to parse.
- * @param  legacyUnit   Legacy unit value, if currentValue needs it appended.
- * @param  units        List of available units.
+ * @param  rawValue   Selected value to parse.
+ * @param  legacyUnit Legacy unit value, if rawValue needs it appended.
+ * @param  units      List of available units.
  *
  * @return A collection of units containing the unit for the current value.
  */
 export function getUnitsWithCurrentUnit(
-	currentValue: Value,
-	legacyUnit: string | undefined,
-	units: WPUnitControlUnitList = ALL_CSS_UNITS
-): WPUnitControlUnitList {
-	if ( ! Array.isArray( units ) ) {
```

Comment:
This function now also expects that `UnitControl` will never be passed a nullish value for its list of units. Is that safe?

Reply:
I've added back the `Array.isArray` check, although I haven't reintroduced the early return  statement — this means that if `units` is null or not an array, the function will still return an array of units (empty in the worst case scenario).

Do you think it's safe enough?

Reply:
Nice, yes that gives us a situation where we're gracefully failing to a desired value. LGTM!